src/hardware/mixer.cpp

```diff
@@ -787,39 +714,27 @@ static void SDLCALL MIXER_CallBack(MAYBE_UNUSED void *userdata, Uint8 *stream, i
 
 	mixer.done -= reduce;
 	mixer.needed -= reduce;
-	pos = mixer.pos;
 	mixer.pos = (mixer.pos + reduce) & MIXER_BUFMASK;
-	if(need != reduce) {
-		while (need--) {
-			Bitu i = (pos + (index >> INDEX_SHIFT_LOCAL )) & MIXER_BUFMASK;
-			index += index_add;
-			sample=mixer.work[i][0]>>MIXER_VOLSHIFT;
-			*output++=MIXER_CLIP(sample);
-			sample=mixer.work[i][1]>>MIXER_VOLSHIFT;
-			*output++=MIXER_CLIP(sample);
-		}
-		/* Clean the used buffer */
-		while (reduce--) {
-			pos &= MIXER_BUFMASK;
-			mixer.work[pos][0]=0;
-			mixer.work[pos][1]=0;
-			pos++;
-		}
-	} else {
-		while (reduce--) {
-			pos &= MIXER_BUFMASK;
-			sample=mixer.work[pos][0]>>MIXER_VOLSHIFT;
-			*output++=MIXER_CLIP(sample);
-			sample=mixer.work[pos][1]>>MIXER_VOLSHIFT;
-			*output++=MIXER_CLIP(sample);
-			mixer.work[pos][0]=0;
-			mixer.work[pos][1]=0;
-			pos++;
-		}
+
+	auto pos = mixer.pos;
+	int idx = 0;
+
+	while (reduce--) {
+		pos &= MIXER_BUFMASK;
+		const MixerFrame frame = {MIXER_CLIP(mixer.work[pos][0] >> MIXER_VOLSHIFT),
+		                    MIXER_CLIP(mixer.work[pos][1] >> MIXER_VOLSHIFT)};
+		queue_buffer[idx++] = frame;
```

Comment:
This is extremely instructive and self-documenting that the left and right samples are making up a frame, which is then added to the queue.  This is very pleasant and easy to read - thank you!